Lib/fontParts/base/base.py

```diff
@@ -52,6 +52,32 @@ def _reprContents(self):
         """
         return []
 
+    # ----
+    # hash
+    # ----
+
+    def __hash__(self):
+        """
+        Subclasses may override this method.
+        """
+        return id(self.naked())
+
+    def __eq__(self, other):
+        """
+        Subclasses may override this method.
+        """
+        if isinstance(other, self.__class__):
+            return self.naked() is other.naked()
+        return NotImplemented
+
+    def __ne__(self, other):
+        """
+        Subclasses may override this method.
+        """
+        if isinstance(other, self.__class__):
+            return not self.naked() is other.naked()
```

Comment:
in python3, `object.__ne__()` automatically delegates to `__eq__()` and inverts the result, unless the latter returns `NotImplemented`. To support both 2 and 3, something like this:

```python
def __ne__(self, other):
    equal = self.__eq__(other)
    return NotImplemented if equal is NotImplemented else not equal
```